streams/jvm/src/main/scala-2.13/zio/stream/ZSinkFromFileConstructor.scala

```diff
@@ -0,0 +1,43 @@
+package zio.stream
+
+import java.nio.ByteBuffer
+import java.nio.channels.FileChannel
+import java.nio.file.StandardOpenOption._
+import java.nio.file.{ OpenOption, Path }
+
+import scala.jdk.CollectionConverters._
+
+import zio._
+import zio.blocking.Blocking
+
+trait ZSinkFromFileConstructor { self: ZSink.type =>
+
+  /**
+   * Uses the provided `Path` to create a [[ZSink]] that consumes byte chunks
+   * and writes them to the `File`. The sink will yield count of bytes written.
+   */
+  final def fromFile(
+    path: => Path,
+    position: Long = 0L,
+    options: Set[OpenOption] = Set(WRITE, TRUNCATE_EXISTING, CREATE)
+  ): ZSink[Blocking, Throwable, Byte, Long] =
+    ZSink {
+      for {
+        state <- Ref.make(0L).toManaged_
+        channel <- ZManaged.make(
+                    blocking.effectBlockingInterrupt(FileChannel.open(path, options.asJava).position(position)).orDie
```

Comment:
Would it be possible to just create the set manually with a loop instead of duplicating the entire file?

Reply:
Yep. It would be better than playing around sbt and crossversion.